src/Service/Manager/SponsorshipLevelManager.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Service\Manager;
+
+use App\Entity\SponsorshipLevel;
+use App\Repository\SponsorshipLevelRepository;
+use App\Service\Assembler\SponsorshipLevelAssembler;
+use App\Service\DTO\SponsorshipLevelData;
+use Doctrine\ORM\EntityManagerInterface;
+
+/**
+ * @method SponsorshipLevelRepository getRepository()
+ */
+final class SponsorshipLevelManager extends EntityManager implements SponsorshipLevelManagerInterface
+{
+    public function __construct(EntityManagerInterface $em, SponsorshipLevelAssembler $assembler)
+    {
+        parent::__construct($em, $assembler, SponsorshipLevel::class);
+    }
+
+    /**
+     * @return SponsorshipLevelData[]
+     */
+    public function getOrderedList()
+    {
+        return $this->getAssembler()->writeMultiple($this->getRepository()->findBy([], ['position' => 'asc']));
+    }
+
+    public function switchPosition(string $move, string $id): SponsorshipLevelData
+    {
+        $sponsorshipLevel = $this->getRepository()->find($id);
+
+        $first_position = $sponsorshipLevel->getPosition();
+        $position = $move == 'left' ? $sponsorshipLevel->getPosition() - 1 : $sponsorshipLevel->getPosition() + 1;
```

Comment:
pre-increment / pre-decrement

Reply:
can't do this here because i need to set :
- Old level with $first_position
- New level with $position
So if i use pre-increment / pre-decrement, my first position is not valid when i want to use it